synapseclient/core/upload/multipart_upload.py

```diff
@@ -472,7 +478,9 @@ def multipart_upload_file(
         mime_type, _ = mimetypes.guess_type(file_path, strict=False)
         content_type = mime_type or 'application/octet-stream'
 
-    md5_hex = md5_for_file(file_path).hexdigest()
+    spinner = Spinner()
+    callback_func = spinner.show_on_terminal if not syn.silent else None
```

Comment:
how about something like below so we don't instantiate the Spinenr unless we use it

progress_callback = Spinner().show_on_terminal if not syn.silent else None